.buildkite/scripts/steps/build_and_test.sh

```diff
@@ -93,7 +93,7 @@ else
          export BOOSTCLANGVER=120
      fi
 
-     (cd ${REPO_ROOT} && ./gradlew --info -Dbuild.version_qualifier=${VERSION_QUALIFIER:-} -Dbuild.snapshot=$BUILD_SNAPSHOT -Dbuild.ml_debug=$ML_DEBUG $TASKS)
+     (cd ${REPO_ROOT} && ./gradlew --info -Dbuild.version_qualifier=${VERSION_QUALIFIER:-} -Dbuild.snapshot=${BUILD_SNAPSHOT:-} -Dbuild.ml_debug=${ML_DEBUG:-} $TASKS)
```

Comment:
Is this intended to say "default is empty string"? That's what the previous code was saying too, because `bash` expands unset environment variables to empty strings.

If you find it much clearer the way it is now then I don't mind leaving it, but personally I found the old way easier to read.

Reply:
> Is this intended to say "default is empty string"?
Yes, that's correct. The script calls `set -u` near the top, which will exit on undefined variables, rather than expanding to the empty string. I do find it much clearer (and much easier to debug) this way, but it is really a matter of taste. I'd prefer to leave it as it is at the moment, mainly because it's now working!

Reply:
Oh, I forgot about the `set -u`. That explains why the explicit default is needed.